lua/ulx/modules/sh/cfc_constraint_checker.lua

```diff
@@ -87,38 +94,63 @@ local function countConstraints( plys )
     return perPlyConstraints
 end
 
-local function printConstraintResults( caller, ply, constraintCounts )
+local function printConstraintResults( _, ply, constraintCounts )
     local decorLength = 25
     local nl = "\n"
-    local divider, nameDivider = string.rep( "=", decorLength ), string.rep( "-", decorLength )
-    local totalCount = "TOTAL: " .. constraintCounts.Total
-    local topBlock = nl .. divider .. nl .. ply:Name() .. "'s constraints:" .. nl .. nameDivider .. nl .. totalCount
-    caller:PrintMessage( 2, topBlock )
+    local divider = string.rep( "=", decorLength ) .. nl
+    local nameDivider = string.rep( "-", decorLength ) .. nl
+    local totalCount = Round( constraintCounts.Total )
+    local totalLabel = "TOTAL: " .. totalCount .. nl
+    local plyTeamColor = team.GetColor( ply:Team() )
+
+    local valueInThreshold = Clamp( totalCount, 0, CONSTRAINT_THRESHOLD )
+    local severity = Remap( valueInThreshold, CONSTRAINT_THRESHOLD, 0, HSV_RED_ANGLE, HSV_GREEN_ANGLE )
+    local colorSeverity = HSVToColor( severity, 1, 0.8 )
+    if totalCount == 0 then colorSeverity = WHITE end
+
+    local blockData = {
+        WHITE, nl .. divider,
+        plyTeamColor, ply:Name(), WHITE, " | ", colorSeverity, totalLabel, WHITE,
+        nameDivider,
+        divider
+    }
 
-    -- Print the rest of the counts
     for constrType, count in pairs( constraintCounts ) do
         if constrType ~= "Total" then
-            caller:PrintMessage( 2, constrType .. ": " .. count )
+            local data = constrType .. ": " .. Round( count ) .. nl
+            table.insert( blockData, #blockData, data )
         end
     end
 
-    caller:PrintMessage( 2, divider )
+    return blockData
 end
 
 
-function cmd.checkConstraints( caller, targetPlys )
+function cmd.checkConstraints( caller, targetPlys, showPlysWithNoConstraints )
     local perPlyConstraints = countConstraints( targetPlys )
+    local dataBlocks = {}
 
     ulx.fancyLogAdmin( caller, true, "#A checked the constraints of #T", targetPlys ) -- Alert staff console of the command being used
 
     for _, ply in pairs( targetPlys ) do
         local constraintCounts = perPlyConstraints[ply]
-
-        printConstraintResults( caller, ply, constraintCounts )
+        if showPlysWithNoConstraints or constraintCounts.Total > 0 then
+            local block = printConstraintResults( caller, ply, constraintCounts )
+            table.Add( dataBlocks, block )
+        end
     end
+
+    net.Start( NETWORK_NAME )
+    net.WriteTable( dataBlocks )
+    net.Send( caller )
+
+    timer.Simple( 0, function()
```

Comment:
Why are you using a zero tick timer here?

Reply:
That is so the print will be after all the other prints, I was told to do this instead of getting into the nitty-gritty of making a callback function from ULX for the command.